src/Uno.UI.RuntimeTests/MUX/Microsoft_UI_Xaml_Controls/TreeViewTests.cs

```diff
@@ -0,0 +1,825 @@
+				selectionChangedEventArgs = null;
+
+				tvi2.IsSelected = false;
+
+				Verify.IsNotNull(selectionChangedEventArgs);
+				Verify.AreEqual(0, selectionChangedEventArgs.AddedItems.Count);
+				Verify.AreEqual(1, selectionChangedEventArgs.RemovedItems.Count);
+				Verify.IsTrue(selectionChangedEventArgs.RemovedItems.Contains(2));
+				selectionChangedEventArgs = null;
+
+				tvi1.IsSelected = true;
+
+				Verify.IsNotNull(selectionChangedEventArgs);
+				Verify.AreEqual(1, selectionChangedEventArgs.AddedItems.Count);
+				Verify.IsTrue(selectionChangedEventArgs.AddedItems.Contains(1));
+				Verify.AreEqual(0, selectionChangedEventArgs.RemovedItems.Count);
+				selectionChangedEventArgs = null;
+
+				treeView.ItemsSource = null;
+
+				Verify.IsNotNull(selectionChangedEventArgs);
+				Verify.AreEqual(0, selectionChangedEventArgs.AddedItems.Count);
+				Verify.AreEqual(1, selectionChangedEventArgs.RemovedItems.Count);
+				Verify.IsTrue(selectionChangedEventArgs.RemovedItems.Contains(1));
+			});
+		}
+
+		[TestMethod]
+		[Ignore("Fails on iOS and WASM")]
+		public async Task TreeViewSelectionChangedMultipleMode()
+		{
+
+			// input data:
+			// - 1
+			//   - 11
+			//   - 12
+			//   - 13
+			// - 2
+			//   - 21
+			// - 3
+			TreeView treeView = null;
+			TreeViewNode node1 = null;
+			TreeViewNode node11 = null;
+			TreeViewNode node12 = null;
+			TreeViewNode node13 = null;
+			TreeViewNode node2 = null;
+			TreeViewNode node21 = null;
+			TreeViewNode node3 = null;
+			TreeViewSelectionChangedEventArgs selectionChangedEventArgs = null;
+			await RunOnUIThread.Execute(() =>
+			{
+				treeView = new TreeView { SelectionMode = TreeViewSelectionMode.Multiple };
+				treeView.SelectionChanged += (s, e) => selectionChangedEventArgs = e;
+
+				node1 = new TreeViewNode { Content = "1", IsExpanded = true };
+				node11 = new TreeViewNode { Content = "11" };
+				node12 = new TreeViewNode { Content = "12" };
+				node13 = new TreeViewNode { Content = "13" };
+				node1.Children.Add(node11);
+				node1.Children.Add(node12);
+				node1.Children.Add(node13);
+
+				node2 = new TreeViewNode { Content = "2", IsExpanded = true };
+				node21 = new TreeViewNode { Content = "21" };
+				node2.Children.Add(node21);
+
+				node3 = new TreeViewNode { Content = "3" };
+
+				treeView.RootNodes.Add(node1);
+				treeView.RootNodes.Add(node2);
+				treeView.RootNodes.Add(node3);
+				TestServices.WindowHelper.WindowContent = treeView;
+			});
+
+			await TestServices.WindowHelper.WaitForIdle();
+
+			await RunOnUIThread.Execute(() =>
+			{
+				var tvi1 = (TreeViewItem)treeView.ContainerFromItem(node1);
+				var tvi11 = (TreeViewItem)treeView.ContainerFromItem(node11);
+				var tvi12 = (TreeViewItem)treeView.ContainerFromItem(node12);
+				var tvi13 = (TreeViewItem)treeView.ContainerFromItem(node13);
+				var tvi2 = (TreeViewItem)treeView.ContainerFromItem(node2);
+				var tvi21 = (TreeViewItem)treeView.ContainerFromItem(node21);
```

Comment:
Codacy found an issue: [Remove the unused local variable 'tvi21'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)